plugins/package-managers/gradle-inspector/src/main/kotlin/GradleInspector.kt

```diff
@@ -286,6 +289,38 @@ private fun Collection<OrtDependency>.toPackageRefs(
         PackageReference(id, linkage, dep.dependencies.toPackageRefs(packageDependencies))
     }
 
+/**
+ * Create a [RemoteArtifact] based on the given [pomUrl], [classifier], [extension] and hash [algorithm]. The hash value
+ * is retrieved remotely.
+ */
+private fun createRemoteArtifact(
+    pomUrl: String?, classifier: String = "", extension: String = "jar", algorithm: String = "sha1"
+): RemoteArtifact {
+    val artifactBaseUrl = pomUrl?.removeSuffix(".pom") ?: return RemoteArtifact.EMPTY
+
+    val artifactUrl = buildString {
+        append(artifactBaseUrl)
+        if (classifier.isNotEmpty()) append("-$classifier")
+        if (extension.isNotEmpty()) append(".$extension")
+    }
+
+    // TODO: How to handle authentication for private repositories here, or rely on Gradle for the download?
+    val checksum = OkHttpClientHelper.downloadText("$artifactUrl.$algorithm")
+        .getOrElse { return RemoteArtifact.EMPTY }
+
+    return RemoteArtifact(artifactUrl, parseChecksum(checksum, algorithm))
+}
+
+/**
+ * Split the provided [checksum] by whitespace and return a [Hash] for the first element that matches the provided
+ * algorithm. If no element matches, return [Hash.NONE]. This works around the issue that Maven checksum files sometimes
+ * contain arbitrary strings before or after the actual checksum.
+ */
+private fun parseChecksum(checksum: String, algorithm: String) =
+    checksum.splitOnWhitespace().firstNotNullOfOrNull {
+        runCatching { Hash.create(it, algorithm) }.getOrNull()
+    } ?: Hash.NONE
```

Comment:
For workarounds like this it would be nice if we could share the code with the Maven implementation somehow later on, because then any improvements will automatically be available for both package managers.

Reply:
I agree. However, I was unsure yet where that code should go, as I actually also have plans to completely rewrite the Maven implementation (see #4720).